Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3900,19 +3901,100 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
         }
     }();
 
-    auto pointInPlugin = convertFromRootViewToPlugin(pointInRootView);
-    auto pointInDocument = convertDown<FloatPoint>(CoordinateSpace::Plugin, CoordinateSpace::PDFDocumentLayout, pointInPlugin);
-    auto pageIndex = m_presentationController->nearestPageIndexForDocumentPoint(pointInDocument);
-    auto pointInPage = convertDown(CoordinateSpace::PDFDocumentLayout, CoordinateSpace::PDFPage, pointInDocument, pageIndex);
+    auto [page, pointInPage] = rootViewToPage(pointInRootView);
+    if (!page)
+        return;
 
-    RetainPtr page = m_documentLayout.pageAtIndex(pageIndex);
     setCurrentSelection([pdfDocument selectionFromPage:page.get() atPoint:pointInPage toPage:page.get() atPoint:pointInPage withGranularity:pdfGranularity]);
 #else
     UNUSED_PARAM(pointInRootView);
     UNUSED_PARAM(granularity);
 #endif
 }
 
+SelectionWasFlipped UnifiedPDFPlugin::moveSelectionEndpoint(WebCore::FloatPoint pointInRootView, SelectionEndpoint extentEndpoint)
+{
+    auto flipped = SelectionWasFlipped::No;
+#if HAVE(PDFDOCUMENT_SELECTION_WITH_GRANULARITY)
+    RetainPtr pdfDocument = m_pdfDocument;
+    if (!pdfDocument)
+        return flipped;
+
+    bool baseIsStart = extentEndpoint == SelectionEndpoint::End;
+    auto baseEndpoint = baseIsStart ? SelectionEndpoint::Start : SelectionEndpoint::End;
+    auto basePageAndPoint = selectionCaretPointInPage(baseEndpoint);
+    if (!basePageAndPoint)
+        return flipped;
+
+    auto extentPageAndPoint = selectionCaretPointInPage(extentEndpoint);
```

Comment:
Is this not used anywhere?

Reply:
🤦 Good catch, this should be removed.

Thank you for the review!